plugin_ide.core.tests/src-lang/melnorme/lang/ide/core/engine/StructureModelTest.java

```diff
@@ -60,8 +60,7 @@ public FixtureSourceModelManager() {
 			super(new DocumentReconcileManager(), new ProblemMarkerUpdater());
 		}
 		
-		public final SourceFileStructure DEFAULT_STRUCTURE = new SourceFileStructure(null, null, 
-			(Indexable<ParserError>) null);
+		public final SourceFileStructure DEFAULT_STRUCTURE = new SourceFileStructure(null, (Indexable<ParserError>) null);
```

Comment:
What do I think? I'm not sure what you mean. Yes there can be StructureElements with different locations, but I don't see what is the issue. Just use the getLocation method as added in https://github.com/RustDT/RustDT/commit/fe73508dce4baba8844e7ac2762440c3fba18ffd Note that for that method to work, you can't clone the StructureElements as you were doing before, but I see you already changed that so that they are not cloned, as suggested.